src/common/cd_image.cpp

```diff
@@ -47,7 +48,8 @@ std::unique_ptr<CDImage> CDImage::Open(const char* filename, Common::Error* erro
   }
   else if (StringUtil::Strcasecmp(extension, ".chd") == 0)
   {
-    return OpenCHDImage(filename, error);
+    const bool precache = g_settings.cdrom_precache_chd && !g_settings.cdrom_load_image_to_ram;
```

Comment:
Common code should not depend on core. The easiest way around this would probably be to add a precache parameter, or better yet, a flags parameter.

Reply:
Is there an existing flags enum somewhere I can extend with a precache field?